### PR TITLE
Search_activity

### PR DESCRIPTION
**Pull request by jerry-peng**
fixes issue #7 #10

### Changes in this PR.

- [X] Added SubredditSearchUtils, a utility class for building URL for subreddit search, and a JSON parsing function.
- [X] Rename the loader class to UrlJsonLoader, for loading JSON data from URL.
- [X] Added SearchActivity, for searching subreddit. Includes a recycler view for all the subreddits searched.
- [X] Added SearchListAdapter, the adapter for the SearchActivity recycler view.
- [X] Add the SQLite DB, Column: name, category, blocked.
- [X] Moved DownloadImageTask to Utils folder
- [X] Added a search button in MainActivity to navigate to SearchActivity

### Testing this PR.

(Require looking at LogCat)
1. Build and load the application
2. Click the search button
3. Enter anything into the text box and click the "search" button
4. See the list of subreddit names/categories and an "add" button pop up. Scroll it to make sure it works.
5. Click on one of the add button, check if Logcat print out that the subreddit is added.
6. Click on the same button again, check if Logcat print out that the subreddit is a duplicate.
7. Go back to MainActivity, and click the search button on the top bar to go to SearchActivity again.
8. Search the same thing as last time, and click on the same SubReddit, check if the duplicate message is printed
9. Search other subreddits and play around.

### Expected Output.
1. The SearchActivity and its recycler view works.
2. The search functions work
3. Clicking on the "add" button on one of the subreddit, the log message shows up.
4. Duplicates are not added, and the duplicate message shows up on log.

## New Fix

- [x] Toast message for adding subreddit.
- [x] Change icon when subreddit added.
- [x] If the subreddit already exists, change icon
- [x] If the subreddit is already added, clicking the icon removes it from DB
- [X] De-cluter the names and categories of subreddit.

### CODE REVIEW

**Comment by morningcoffeexs** [comment]
Yess good work on this pull request Jerry! This works well on my end, but there are some things I'd like us to fix after this merge:

1. Toast message to notify the user that the subreddit in the search activity has been added upon click
2. Change the icon of the add button to a checkmark upon click
3. Include an "Subreddit already exists" boolean in the view holder for each search item. If this boolean is true, then change the add button icon to a checkmark. 
4. Remove the categories text on the right of the subreddit name. This creates clutter in the search, and isn't necessary as the subreddit name should already tell the user enough. By including categories, we will also be adding more issues to deal with (in this case, too many categories to list: https://puu.sh/zN4aK/c2c1d17973.png)

I'll wait for @jawhnypoh to view this pull request and merge it himself. Good work once again!

**Comment by jerry-peng** [comment]
@trilxs Issues stated above fixed.
Changes:
If subreddit exists in database, change to "check" icon.
If user click on the "check" icon, removes the subreddit from database, change icon back to "add" icon
Toast message appears.
If subredditName too long, minimize with "..."
For categories, only get string before the comma, and if string still too long, minimize with "..."

**Review by jawhnypoh — APPROVED** [review]
++

**Pull request merged**
